netket/callbacks/invalidlossstopping.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020, 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Union
+
+import numpy as np
+
+
+@dataclass
+class InvalidLossStopping:
+    """A simple callback to stop NetKet if there are no more improvements in the training.
+    based on `driver._loss_name`."""
+
+    monitor: str = "mean"
+    """Loss statistic to monitor. Should be one of 'mean', 'variance', 'sigma'."""
+    patience: Union[int, float] = 0
+    """Number of epochs with invalid loss after which training will be stopped."""
+
+    def __post_init__(self):
+        self._last_valid_iter = 0
+
+    def __call__(self, step, log_data, driver):
+        """
+        A boolean function that determines whether or not to stop training.
+
+        Args:
+            step: An integer corresponding to the step (iteration or epoch) in training.
+            log_data: A dictionary containing log data for training.
+            driver: A NetKet variational driver.
+
+        Returns:
+            A boolean. If True, training continues, else, it does not.
+        """
+        loss = np.real(getattr(log_data[driver._loss_name], self.monitor))
+        if np.isnan(loss):
```

Comment:
```suggestion
        if np.isnan(loss):
```
I wonder if `np.isfinite` would be better? It tests for `NaN`, but also for `±Infinity`, which also seems like a problematic expectation that might warrant aborting the simulation.

What do you think? @VolodyaCO @PhilipVinc

Reply:
Sure thing!
```suggestion
        if np.isfinite(loss):
```